assets/js/custom/TranslateProgram.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-env jquery */
+/* global Translation */
+
+// eslint-disable-next-line no-unused-vars
+class TranslateProgram extends Translation {
+  constructor (translatedByLine, programId, hasDescription, hasCredit) {
+    super(translatedByLine)
+    this.programId = programId
+    this.hasDescription = hasDescription
+    this.hasCredit = hasCredit
+    this._initListeners()
+  }
+
+  _initListeners () {
+    const self = this
+    $(document).on('click', '#program-translation-button', function () {
+      $(this).hide()
+
+      if (self.isTranslationNotAvailable('#name-translation')) {
+        $('#program-translation-loading-spinner').show()
+        self.translateProgram()
+      } else {
+        self.openTranslatedProgram()
+      }
+    })
+
+    $(document).on('click', '#remove-program-translation-button', function () {
+      $(this).hide()
+      $('#program-translation-button').show()
+
+      $('#name').removeClass('program-name').addClass('program-name-animation')
+      $('#name-translation').removeClass('program-name').addClass('program-name-animation')
+      $('#name-translation').animate({ width: 'toggle' })
+      $('#name').animate({ width: 'toggle' }, 400,
+        function () {
+          $('#name').removeClass('program-name-animation').addClass('program-name')
+          $('#name-translation').removeClass('program-name-animation').addClass('program-name')
+        }
+      )
+
+      if (self.hasDescription) {
+        $('#description').slideDown()
+        $('#description-translation').slideUp()
+      }
+
+      $('#credits-translation-wrapper').slideUp()
```

Comment:
This function will always remove the credits-translation-wrapper since it holds the program credits and translation credits where the description will only be closed when there is a description in the program. Also, the naming convention is different since it is a wrapper that holds both the program and translation credits